### ISSUE TITLE
JMAP: Support "createdIds" in method call and response objects

### ISSUE THREAD

**Issue by robmueller** [cyrusimap/cyrus-imapd#2460]
cyrus doesn't currently seem to support the `createdIds` method call parameter and response object. We really need this so the perl middleware proxy can work in all cases.

https://jmap.io/spec-core.html#the-request-object

The request object:

```
createdIds: String[String] (optional) A map of (client-specified) creation id to the id the server assigned when a record was successfully created.

As described later in this specification, some records may have a property that contains the id of another record. To allow more efficient network usage, you can set this property to reference a record created earlier in the same API request. Since the real id is unknown when the request is created, the client can instead specify the creation id it assigned, prefixed with a #. As the server processes API requests, any time it successfully creates a new record it adds to this map the creation id, with the server-assigned real id as the value. If it comes across a reference to a creation id in a create/update, it looks it up in the map and replaces the reference with the real id, if found.

The client can pass an initial value for this map as the createdIds property of the Request. This MAY be an empty object. If given in the request, the response will also include a createdIds property, with any additionally created ids added. This allows proxy servers to easily split a JMAP request into multiple JMAP requests to send to different servers. For example it could send the first two method calls to server A, then the third to server B, before sending the fourth to server A again. By passing the createdIds of the previous response to the next request, it can ensure all of these still resolve.
```

The response object:

```
createdIds: String[String] (optional; only returned if given in request) A map of (client-specified) creation id to the id the server assigned when a record was successfully created. This includes all values passed in the request, as well as any additional ones added for newly created records.
```

**Comment by rsto** [comment]
We are asserting `createdIds` in this test: https://github.com/cyrusimap/cassandane/blob/master/Cassandane/Cyrus/JMAPCore.pm#L231

Is that not working for you or is there a gap with Cyrus to the spec?

**Comment by rsto** [comment]
Fixed in 6002156a3134b534e752188e38da90c55fdd928f
